jquery/jquery.d.ts

```diff
@@ -998,7 +998,22 @@ interface JQueryStatic {
      * @param objectN Additional objects containing properties to merge in.
      */
     extend(deep: boolean, target: any, object1?: any, ...objectN: any[]): any;
-
+    /**
+     * Merge the contents of two or more objects together into the first object.
+     *
+     * @param target An object that will receive the new properties if additional objects are passed in or that will extend the jQuery namespace if it is the sole argument.
+     * @param objects Rest parameter for use with spread operator for array of objects containing properties to merge in. 
+     */
+    extend(target: any, ...objects: any[]): any;
+    /**
+     * Merge the contents of two or more objects together into the first object.
+     *
+     * @param deep If true, the merge becomes recursive (aka. deep copy).
+     * @param target The object to extend. It will receive the new properties.
+     * @param objects Rest parameter for use with spread operator for array of objects containing properties to merge in.
+     */
+    extend(deep: boolean, target: any, ...objects: any[]): any;
```

Comment:
@basarat Oops, I did not mean to add the additional commits to this pr.

I was trying to use it directly with a spread.

```
            var test: {}[];
            return $.extend({}, ...test);
```

This would have a runtime exception if test was undefined. Although a different exception, passing undefined as object1 would also have result in an exception.